miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -161,6 +163,56 @@ proc.update_account_code
     # => []
 end
 
+
+# ACCOUNT STORAGE UPDATE
+# =================================================================================================
+
+#! Updates the account storage commitment by hashing the storage slots and setting the storage
+#! commitment to the newly computed storage hash.
+#!
+#! Stack: []
+#! Output: []
+proc.update_account_storage
+    # get number of storage slots
+    exec.memory::get_num_storage_slots
+    # => [num_storage_slots]
+
+    # check if there are storage slots
+    push.0 neq
+    # => [storage_slots_is_not_empty]
+
+    # only loop and hash over storage if there are storage slots
+    if.true
+        # get num storage slots
+        exec.memory::get_num_storage_slots
+        # => [num_storage_slots]
```

Comment:
Instead of making another call to `get_num_storage_slots`, I would just `dup` the value we got on line 177.